qiskit/circuit/quantumcircuit.py

```diff
@@ -1910,41 +1929,56 @@ def assign_parameters(self, param_dict, inplace=False):
         # replace in self or in a copy depending on the value of in_place
         bound_circuit = self if inplace else self.copy()
 
-        # unroll the parameter dictionary (needed if e.g. it contains a ParameterVector)
-        unrolled_param_dict = self._unroll_param_dict(param_dict)
+        if isinstance(params, dict):
+            # unroll the parameter dictionary (needed if e.g. it contains a ParameterVector)
+            unrolled_param_dict = self._unroll_param_dict(params)
 
-        # check that all param_dict items are in the _parameter_table for this circuit
-        params_not_in_circuit = [param_key for param_key in unrolled_param_dict
-                                 if param_key not in self.parameters]
-        if len(params_not_in_circuit) > 0:
-            raise CircuitError('Cannot bind parameters ({}) not present in the circuit.'.format(
-                ', '.join(map(str, params_not_in_circuit))))
-
-        # replace the parameters with a new Parameter ("substitute") or numeric value ("bind")
-        for parameter, value in unrolled_param_dict.items():
-            bound_circuit._assign_parameter(parameter, value)
+            # check that all param_dict items are in the _parameter_table for this circuit
+            params_not_in_circuit = [param_key for param_key in unrolled_param_dict
+                                     if param_key not in self.parameters]
+            if len(params_not_in_circuit) > 0:
+                raise CircuitError('Cannot bind parameters ({}) not present in the circuit.'.format(
+                    ', '.join(map(str, params_not_in_circuit))))
 
+            # replace the parameters with a new Parameter ("substitute") or numeric value ("bind")
+            for parameter, value in unrolled_param_dict.items():
+                bound_circuit._assign_parameter(parameter, value)
+        else:
+            if len(params) != self.num_parameters:
+                raise ValueError('Mismatching number of values and parameters. For partial binding '
+                                 'please pass a dictionary of {parameter: value} pairs.')
+            for i, value in enumerate(params):
+                bound_circuit._assign_parameter(self._parameter_table[i], value)
         return None if inplace else bound_circuit
 
-    def bind_parameters(self, value_dict):
+    @deprecate_arguments({'value_dict': 'values'})
+    def bind_parameters(self, values, *, value_dict=None):  # pylint: disable=unused-argument
```

Comment:
why is the `*` necessary?

Reply:
To swallow all positional arguments a user might wrongly throw at the function. I essentially copied this from when we deprecated the positional argument `q` in favor of `qubit` in the circuit gates.

Reply:
I removed the `*` 🙂